JoinColony/colonyDapp
# Motions and Disputes Reveal Vote Widget

## Description

This PR adds in the voting reveal widget to the motion page, but does not wire in the saga

This widget has two states:
1. vote was not revealed yet, just show a info message
2. vote was revealed, show the user's vote

**Notes on testing**
To test this properly you'll need to 
- get two users in the colony
- make sure both have 50% reputation (eg: make a payment of 1000 to each user)
- create the motion
- stake both sides (you can do this with just one user)
- vote (one user for yes, another one for no)
- _at this point you can see the initial state of the reveal widget_
- reveal one user's vote
- _now you can see the vote, of the user who revealed_
- _log in with the user who didn't reveal to make sure that it doesn't show a revealed vote_

**Changes** 
- [x] Extract `VoteDetails` into separate component
- [x] Added `RevealWidget` Component
- [x] Added `motionUserVoteRevealed` client resolver
- [x] Added `MotionVote` enum type
- [x] Fixed `getMotionState` util

**Demo**

Both states, voting not revealed yet (first motion), vote revealed (second motion)

![demo-md-reveal-widget-1](https://user-images.githubusercontent.com/1193222/115291719-c2f81200-a15d-11eb-974a-6cd533a1cbba.gif)

Vote not revealed user logged out / logged in

![demo-md-reveal-widget-2](https://user-images.githubusercontent.com/1193222/115291724-c390a880-a15d-11eb-88f8-5772690bc771.gif)

Vote revealed user logged out / logged in

![demo-md-reveal-widget-3](https://user-images.githubusercontent.com/1193222/115291725-c4293f00-a15d-11eb-9fec-c69eded0e48a.gif)

User not logged in:
![Screenshot from 2021-04-21 13-24-12](https://user-images.githubusercontent.com/1193222/115538914-0ad58100-a2a5-11eb-8bcc-416cb979e84d.png)

User logged in, who VOTED
![Screenshot from 2021-04-21 13-24-17](https://user-images.githubusercontent.com/1193222/115538946-13c65280-a2a5-11eb-99ae-9e30a6b61195.png)

User logged in, didn't VOTE
![Screenshot from 2021-04-21 13-24-24](https://user-images.githubusercontent.com/1193222/115538969-1aed6080-a2a5-11eb-9b9e-6358e02dc0ca.png)

Resolves DEV-288
